packages/lodestar/src/db/api/beacon/repositories/merkleTree.ts

```diff
@@ -0,0 +1,21 @@
+import {FullDatabaseRepository} from "../repository";
+import {MerkleTree} from "../../../../types";
+import {IBeaconConfig} from "../../../../config";
+import {IDatabaseController} from "../../../controller";
+import {Bucket} from "../../../schema";
+import {IProgressiveMerkleTree, ProgressiveMerkleTree} from "../../../../util/merkleTree";
+
+export class MerkleTreeRepository extends FullDatabaseRepository<MerkleTree> {
+
+  public constructor(
+    config: IBeaconConfig,
+    db: IDatabaseController) {
+    super(config, db, Bucket.merkleTree, config.types.MerkleTree);
+  }
+
+  public async getProgressiveMerkleTree(index: number): Promise<IProgressiveMerkleTree> {
```

Comment:
can you extend `get` here?

Maybe IProgressiveMerkleTree can extend MerkleTree so it can be extended.

Reply:
Tried , unfortunately you cannot modify interface or return type of method you are inheriting